sys/mips/mips/trap.c

```diff
@@ -1156,6 +1191,15 @@ trap(struct trapframe *trapframe)
 		if (KTRPOINT(td, KTR_CEXCEPTION))
 			ktrcexception(trapframe);
 #endif
+
+		if (try_emulate_capdirty(trapframe, NULL,
```

Comment:
Worth noting the KTRACE stuff above, but I think it makes sense to trace the CP2 exceptions even for capdirty.

Reply:
Hm.  `pmap_emulate_modified` get called without tracepoint opportunity (see `T_TLB_MOD`), but CoW faults are processed through `vm_fault_trap` which has a tracepoint right at its top.  I could go either way about how committed we are to transparency of capdirty (and CLG) faults.